src/System.Net.Mime/src/System/Net/Mail/AlternateViewCollection.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.ObjectModel;
+
+namespace System.Net.Mail
+{
+    public sealed class AlternateViewCollection : Collection<AlternateView>, IDisposable
+    {
+        private bool _disposed = false;
+
+        internal AlternateViewCollection()
+        { }
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            foreach (AlternateView view in this)
```

Comment:
Not necessarily for this PR, but a follow-up improvement would be to explicitly pass a new `List<AlternateView>` to the base `Collection<T>` constructor from this type's constructor , and then cast `Items` to `List<AlternateView>` here to enumerate the collection more efficiently (avoiding the enumerator allocation and overhead).